src/Core/Checkout/Document/Renderer/CreditNoteRenderer.php

```diff
@@ -199,39 +199,48 @@ public function getDecorated(): AbstractDocumentRenderer
         throw new DecorationPatternException(self::class);
     }
 
-    private function getOrder(string $orderId, string $versionId, Context $context, string $deepLinkCode = ''): OrderEntity
+    private function getOrder(DocumentGenerateOperation $operation, string $versionId, Context $context, DocumentRendererConfig $rendererConfig): OrderEntity
     {
-        ['language_id' => $languageId] = $this->getOrdersLanguageId([$orderId], $versionId, $this->connection)[0];
-
-        // Get the correct order with versioning from reference invoice
-        $versionContext = $context->createWithVersionId($versionId)->assign([
-            'languageIdChain' => array_values(array_unique(array_filter([$languageId, ...$context->getLanguageIdChain()]))),
-        ]);
-
-        $criteria = OrderDocumentCriteriaFactory::create([$orderId], $deepLinkCode, self::TYPE)
-            ->addFilter(new EqualsFilter('lineItems.type', LineItem::CREDIT_LINE_ITEM_TYPE));
+        ['language_id' => $languageId] = $this->getOrdersLanguageId([$operation->getOrderId()], $versionId, $this->connection)[0];
+        $languageIdChain = array_values(
+            array_unique(
+                array_filter([$languageId, ...$context->getLanguageIdChain()])
+            )
+        );
 
-        /** @var ?OrderEntity $order */
-        $order = $this->orderRepository->search($criteria, $versionContext)->get($orderId);
+        // First try to load the order with the version from the reference invoice
+        $order = $this->loadOrder($operation, $versionId, $context, $languageIdChain, $rendererConfig)
+            ?? $this->loadOrder($operation, Defaults::LIVE_VERSION, $context, $languageIdChain, $rendererConfig);
 
-        if ($order) {
-            return $order;
+        if ($order === null) {
+            throw DocumentException::orderNotFound($operation->getOrderId());
         }
 
-        $versionContext = $context->createWithVersionId(Defaults::LIVE_VERSION)->assign([
-            'languageIdChain' => array_values(array_unique(array_filter([$languageId, ...$context->getLanguageIdChain()]))),
-        ]);
+        return $order;
+    }
 
-        $criteria = OrderDocumentCriteriaFactory::create([$orderId], $deepLinkCode, self::TYPE);
+    private function loadOrder(
+        DocumentGenerateOperation $operation,
+        string $versionId,
+        Context $context,
+        array $languageIdChain,
+        DocumentRendererConfig $rendererConfig,
+    ): ?OrderEntity {
+        $versionContext = $context->createWithVersionId($versionId)->assign([
+            'languageIdChain' => $languageIdChain,
+        ]);
 
-        /** @var ?OrderEntity $order */
-        $order = $this->orderRepository->search($criteria, $versionContext)->get($orderId);
+        $criteria = OrderDocumentCriteriaFactory::create([$operation->getOrderId()], $rendererConfig->deepLinkCode, self::TYPE)
+            ->addFilter(new EqualsFilter('lineItems.type', LineItem::CREDIT_LINE_ITEM_TYPE));
```

Comment:
Regarding the failing test, I would fix it like this
```suggestion

        $lineItemAssociation = $criteria->getAssociation('lineItems');
        $lineItemAssociation->addFilter(new EqualsFilter('type', LineItem::CREDIT_LINE_ITEM_TYPE));
```

However this would mean, that the order which is loaded with the version ID from the invoice, would be loaded, if a credit line item is present, and hence not the default version of the order would be loaded. But I would think this is the correct behavior.

Then in line 120 an error would be triggered, that no credit line items are found.

The current tests are working with that change, but as this would be an additional change in behavior, I thought I would mention it.

Reply:
A change in behavior would be better mentioned in the changelog 😉 .

But I don't really get why some renders care about the order version id and others don't. Why does it matter for some? Is a new order version created when e.g. an invoice is generated, but the invoice renderer is using the live version or not? Maybe someone can enlighten me 🙈 @En0Ma1259 @mstegmeyer